estate/models/estate_property.py

```diff
@@ -0,0 +1,105 @@
+    _name = 'estate.property'
+    _description = "Real Estate Property"
+    _order = "id desc"
+
+    name = fields.Char(required=True)
+    description = fields.Text()
+    postcode = fields.Char()
+    date_availability = fields.Date(copy=False, default=fields.Date.today() + relativedelta(months=3))
+    expected_price = fields.Float(required=True, default=0)
+    selling_price = fields.Float(readonly=True, copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Float(string="Living Area (sqm)")
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Float(string="Garden Area (sqm)")
+    garden_orientation = fields.Selection(
+        string="Garden Orientation",
+        selection=[
+        ('north', 'North'),
+        ('south', 'South'),
+        ('east', 'East'),
+        ('west', 'West')
+        ]
+    )
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+        selection=[
+            ('new', 'New'),
+            ('offer_received', 'Offer Received'),
+            ('offer_accepted', 'Offer Accepted'),
+            ('sold', 'Sold'),
+            ('cancelled', 'Cancelled')
+        ],
+        default="new"
+    )
+    property_type_id = fields.Many2one("estate.property.type")
+    buyer_id = fields.Many2one("res.partner", copy=False)
+    salesperson_id = fields.Many2one("res.users", default=lambda self: self.env.user)
+    tag_ids = fields.Many2many("estate.property.tag")
+    offer_ids = fields.One2many("estate.property.offer", "property_id", string="Offers")
+    total_area = fields.Float(compute="_compute_total_area")
+    best_offer = fields.Float(compute="_compute_best_offer")
+
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends('offer_ids.price')
+    def _compute_best_offer(self):
+        for record in self:
+            if not record.mapped('offer_ids.price'):
+                record.best_offer = 0
+            else:
+                record.best_offer = max(record.mapped('offer_ids.price'))
+
+    @api.onchange('garden')
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = 'north'
+        else:
+            self.garden_area = 0
+            self.garden_orientation = False
+
+    def cancel_property(self):
+        for record in self:
+            if record.state == 'sold':
+                raise UserError("sold property cannot be cancelled.")
+            else:
+                record.state = 'cancelled'
+
+    def sold_property(self):
+        for record in self:
+            if record.state == 'cancelled':
+                raise UserError("cancelled property cannot be sold.")
+            else:
+                record.state = 'sold'
+
+    _check_price = models.Constraint(
```

Comment:
The constraint should be defined after the field declaration. You can take an example from Odoo code.